src/objective/regression_obj.cu

```diff
@@ -187,6 +185,31 @@ class RegLossObj : public FitInterceptGlmLike {
         .Eval(io_preds);
   }
 
+  void InitEstimation(MetaInfo const& info, linalg::Vector<float>* base_score) const override {
+    linalg::Vector<float> out;
+    FitInterceptGlmLike::InitEstimation(info, &out);
+    if (std::abs(this->param_.scale_pos_weight - 1.0f) <= kRtEps) {
+      common::Mean(this->ctx_, out, base_score);
+      return;
+    }
+
+    // Special handling for the scale_pos_weight parameter
+    auto w = this->param_.scale_pos_weight;
```

Comment:
Since these are scalar calculations, they could be done in higher precision (fp64 or even 'long double') without loss of speed.

Reply:
I'm leaning toward using one-step newton instead. I don't want too much workaround for a single parameter in the initialization step.

Reply:
I would say the mean initialization approach is quite valuable in other important ways too.

For example, in the issue, it would make the model have an expected value of $\mathbb{E}[y] = 0.5$ by design, which would not be the case with a one-step Newton initialization.